driver/config/tls.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"crypto/tls"
+
+	"github.com/ory/x/tlsx"
+)
+
+const (
+	KeySuffixTLSStrict               = "tls.strict"
+	KeySuffixTLSAllowTerminationFrom = "tls.allow_termination_from"
+	KeySuffixTLSCertString           = "tls.cert.base64"
+	KeySuffixTLSKeyString            = "tls.key.base64"
+	KeySuffixTLSCertPath             = "tls.cert.path"
+	KeySuffixTLSKeyPath              = "tls.key.path"
+
+	KeyTLSAllowTerminationFrom = "serve." + KeySuffixTLSAllowTerminationFrom
+	KeyTLSCertString           = "serve." + KeySuffixTLSCertString
+	KeyTLSKeyString            = "serve." + KeySuffixTLSKeyString
+	KeyTLSCertPath             = "serve." + KeySuffixTLSCertPath
+	KeyTLSKeyPath              = "serve." + KeySuffixTLSKeyPath
+)
+
+type TLSConfig interface {
+	Strict() bool
+	AllowTerminationFrom() []string
+	Certificate() ([]tls.Certificate, error)
+}
+
+func (p *Provider) TLS(iface ServeInterface) TLSConfig {
+	strict := false
+	if !p.forcedHTTP() {
+		strict = p.p.Bool(iface.Key(KeySuffixTLSStrict))
+	}
```

Comment:
I would change this so that:

1. Public can have TLS disabled if forcedHTTP is true, otherwise it is enabled - hence remove `public.tls.strict`
2. Admin can have TLS disabled if forcedHTTP is true, or if `admin.tls.enabled` is `false`.